src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DampedHarmonic.cpp

```diff
@@ -257,39 +279,66 @@ void damped_harmonic_impl(
     get<0>(d4_RW_L2) += get(weight) * d0_roll_on;
   }
 
+  time_deriv_of_lapse<DataVector, SpatialDim, Frame>(
+      make_not_null(&dt_lapse), lapse, shift, spacetime_unit_normal, phi, pi);
+  spatial_deriv_of_lapse<DataVector, SpatialDim, Frame>(
+      make_not_null(&d3_lapse), lapse, spacetime_unit_normal, phi);
+
+  get<0>(d_lapse_by_lapse) = get(one_over_lapse) * get(dt_lapse);
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    d_lapse_by_lapse.get(i + 1) = get(one_over_lapse) * d3_lapse.get(i);
+  }
+  get(det_spatial_metric) = square(get(sqrt_det_spatial_metric));
+  spacetime_deriv_of_det_spatial_metric<DataVector, SpatialDim, Frame>(
+      make_not_null(&d_g_by_det), sqrt_det_spatial_metric,
+      inverse_spatial_metric, d0_spatial_metric, phi);
+  for (size_t a = 0; a < SpatialDim + 1; ++a) {
+    d_g_by_det.get(a) = d_g_by_det.get(a) / get(det_spatial_metric);
+  }
+  const auto spacetime_deriv_of_power_log_factor_metric_lapse =
+      [&d_lapse_by_lapse, &d_g_by_det, &lapse, &sqrt_det_spatial_metric,
+       &prefac_log, &d_logfac,
+       &logfac](gsl::not_null<tnsr::a<DataVector, SpatialDim, Frame>*> result,
+                double g_exponent, int exponent) -> void {
+    for (size_t a = 0; a < SpatialDim + 1; ++a) {
+      d_logfac.get(a) =
+          g_exponent * d_g_by_det.get(a) - d_lapse_by_lapse.get(a);
+    }
+    DampedHarmonicGauge_detail::log_factor_metric_lapse(
+        make_not_null(&logfac), lapse, sqrt_det_spatial_metric, g_exponent);
+    get(prefac_log) =
+        static_cast<double>(exponent) * integer_pow(get(logfac), exponent - 1);
+    for (size_t a = 0; a < SpatialDim + 1; ++a) {
+      result->get(a) = get(prefac_log) * d_logfac.get(a);
+    }
+  };
   // \partial_a \mu_{S} = \partial_a(A_S R_S W
   //                               \log(\sqrt{g}/N)^{c_{S}})
   // \partial_a \mu_1 = \partial_a(A_L1 R_L1 W
   //                               \log(\sqrt{g}/N)^{1+c_{L1}})
   // \partial_a \mu_2 = \partial_a(A_L2 R_L2 W
   //                               \log(1/N)^{1+c_{L2}})
-  DampedHarmonicGauge_detail::spacetime_deriv_of_power_log_factor_metric_lapse<
```

Comment:
Can these detail functions be deleted now?

Reply:
yes, I removed them in a new commit.